db/db.go

```diff
@@ -950,9 +950,10 @@ func (db *DB) queryStmtWithConn(ctx context.Context, stmt *command.Statement, xT
 	}
 	needsQueryTypes := containsEmptyType(xTypes)
 
+	mem := make([]interface{}, len(columns)*2) // single allocation for both dest and ptrs
```

Comment:
This makes sense to me, but this might be *too* clever. Instead just perform two allocations, and the code is clearer. The difference is time is small, especially if amortized over many loops.

Reply:
ok